src/core/src/poet2_util.rs

```
@@ -36,8 +36,27 @@ pub fn blockid_to_hex_string(blockid: BlockId) -> String {
     to_hex_string(&blockid_vec)
 }
 
-pub fn payload_to_wc_and_sig(payload: &Vec<u8>)
-                             -> (String, String) {
+pub fn format_block( block: &Block ) -> String {
+    format!("{}{}{}{}{}{}{}{}{}",
+                   String::from("Block [ block_num : "),
+                   block.block_num,
+                   String::from(", block_id : "),
+                   to_hex_string(&block.block_id),
+                   String::from(", prev_block_id : "),
+                   to_hex_string(&block.previous_id),
+                   String::from(", peer_id : "),
+                   to_hex_string(&block.signer_id),
+                   String::from(" ] "),
+    )
+
+}
+
+pub fn payload_to_wc_and_sig(payload: &Vec<u8>) 
+    -> (String, String) {
+    if payload.len() == 0 {
```

Empty payload is an error condition. Either throw an error or let it panic

---

Clippy warning: We can use payload.is_empty() instead of  comparing array.len() == 0